openpype/plugins/publish/integrate_hero_version.py

```diff
@@ -306,6 +306,23 @@ def integrate_instance(
                 anatomy_filled = anatomy.format(anatomy_data)
                 template_filled = anatomy_filled[template_key]["path"]
 
+                # For representations that have the same extension, an
+                # additional suffix can be available to make the destination
+                # filename different.
+                hero_suffix = repre_info["representation"]["data"].get(
+                    "heroSuffix"
+                )
+                if hero_suffix:
+                    fill_data = copy.deepcopy(template_filled.used_values)
+                    template_filled.template = template_filled.replace(
+                        "." + fill_data["ext"],
+                        "_{}.{}".format(hero_suffix, fill_data["ext"])
+                    )
+                    template_filled = StringTemplate(
+                        template_filled.template
+                    ).format(fill_data)
```

Comment:
Interesting - shouldn't this just be using the `name` of the representation then? Since that's required to be unique anyway?

Reply:
Good point, would still need to enforce the name to be in the template cause current default settings do not have it in there.

Reply:
Actually it turns out the logic was not needed. The hero integration plugin was failing because its was checking for overwriting files, which happened when the alembic and gpu cache wanted to be the same file name. Since using `"outputName": "gpu_cache"` on the representation, the output file name is unique and the hero integration plugin succeeds.
Problem is actually with the integration plugin, which does not fail when trying to write to the same file. Have setup an issue for this https://github.com/ynput/OpenPype/issues/4656